src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,189 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    public MyLinkedList() {
+        this.first = null;
+        this.last = null;
+        this.size = 0;
+    }
+
+    private void validIndex(int index) {
+        if (index < 0 || index >= this.size) {
+            throw new ArrayIndexOutOfBoundsException("Illegal Index: " + index);
+        }
+    }
+
+    private Node<T> indexedNode(int index) {
+        Node<T> node;
+        if (index < this.size / 2) {
+            node = this.first;
```

Comment:
```suggestion
            node = first;
```
let's remove `this` everywhere

Reply:
done